src/accounts.rs

```diff
@@ -291,6 +292,44 @@ impl Accounts {
         }
     }
 
+    /// Performs a background fetch for all accounts in parallel.
+    ///
+    /// If you need a timeout, then use [Accounts::background_fetch_with_timeout] instead.
+    ///
+    /// The `AccountsBackgroundFetchDone` event is emitted at the end,
+    /// process all events until you get this one and you can safely return to the background
+    /// without forgeting to create notifications caused by timing race conditions.
+    pub async fn background_fetch(&self) {
```

Comment:
this function seems not to be needed as public; and making it public overcomplicates things:

apart from documentation and maintainance effort, eg. the `AccountsBackgroundFetchDone` event seems to be emitted twice (`background_fetch_with_timeout` calls `background_fetch()` and both emit `AccountsBackgroundFetchDone` at the end)

Reply:
I See the following possibilities:
A. only emit the event in `background_fetch_with_timeout`
B. conditionally emit the event in `background_fetch_with_timeout` only on timeout
C. merge both functions into one
D. same as C, but keep it in two functions, making the `background_fetch` private / internal

As this is about networking, and it is in the nature of networking to be flaky, I'd say there is no really a use case for calling the background fetch without a timeout, so I think we should go for C (or D if we think we will find a use case for reusing the function in the future).

Reply:
yip, let's go for C. we currently do no need the case of "no timeout" and this is also not forseeable. yagni :)